Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp

```diff
@@ -1037,61 +1066,62 @@ void WebChromeClient::setNeedsOneShotDrawingSynchronization()
 
 bool WebChromeClient::shouldTriggerRenderingUpdate(unsigned rescheduledRenderingUpdateCount) const
 {
-    return m_page.shouldTriggerRenderingUpdate(rescheduledRenderingUpdateCount);
+    return protectedPage()->shouldTriggerRenderingUpdate(rescheduledRenderingUpdateCount);
 }
 
 void WebChromeClient::triggerRenderingUpdate()
 {
-    if (m_page.drawingArea())
-        m_page.drawingArea()->triggerRenderingUpdate();
+    if (auto* drawingArea = page().drawingArea())
```

Comment:
Why don't we need to protect the page or the drawing area here?

Reply:
DrawingArea is not refcounted but I'll protect the page in local variable.